pkg/crosscluster/replicationutils/utils.go

```diff
@@ -260,6 +262,65 @@ func ResolveHeartbeatTime(
 	return newProtectAbove
 }
 
+func GetAlterConnectionChecker(
```

Comment:
nit: consider putting this in its own file. My rule of thumb is once something crosses 50 lines, its big enough to get its own file.

Reply:
done.